### DISCUSSION TITLE
Onload body py attribut event

### DISCUSSION THREAD

**Discussion by frchalaoux** [pyscript/pyscript#1185] [Q&A] [answered]
Hi, is it possible to call a python fonction with the Onload body py attribut event like as written below ?
For the moment it does not work for me either in html or ! Any suggestion ?

`<body py-load="refresh()" id=mydocument>
`
or  directly in 

```
<py-script defer src="./cms.py">
</py-script>
```

```
def refresh():
blablabla

def main():
doc= document.querySelector(“#mydocument”)
doc.addEventListener(“load”, create_proxy(refresh))
```

Total code is available at [My repository](https://github.com/frchalaoux/cms-pyscript/tree/Bug-publish1-pyscript)

**Comment by JeffersGlass** [comment, marked as answer]
I believe the issue here is that the native `load` event dispatches before the Python interpreter has been initialized, and before any PyScript event handlers have been installed. For example, try the following code and open the developer console log:
```
<!doctype html>
<html>
<head>
<script defer src="https://pyscript.net/releases/2022.12.1/pyscript.js"></script>
<link rel="stylesheet" href="https://pyscript.net/releases/2022.12.1/pyscript.css"/>
</head>
<body> 
  <script>
    window.addEventListener("load", (event) => {
      console.warn("page is fully loaded");
    });
  </script>   
<py-script>
  print("Hello, world")
</py-script>
</body>
</html>
```
![image](https://user-images.githubusercontent.com/1931111/218505395-0bbe8fcb-6806-4ec6-ae19-69e451cbba74.png)
At least for me, the `load` event is dispatched to the window just after the Pyodide interpreter is downloaded or fetched from cache, before it's even initialized in the browser. So PyScript's event handler's haven't been installed yet, and can't catch it.

**Reply by frchalaoux** [reply]
Thank Jeffers for your answer !

Just call refresh() in the main and it works !

**Reply by JeffersGlass** [reply]
Sounds good! Would it be ok to mark this question as `answered`?

**Reply by frchalaoux** [reply]
Done !
